### PR TITLE
Add properties for getting Result's value

### PR DESCRIPTION
**Pull request by adasiewiczr**
### 🎯 Goal

Add `Result::getOrNull`, `Result::getOrThrow` and `Result::chatErrorOrNull`.

### 🛠 Implementation details

1. Added two properties to the `Result` class to simplify getting `value` for Java users
2. Converted `onSuccess` and `onError` from extension functions to methods
3. Removed `@JvmSynthetic` from some result's method

### 🧪 Testing

Just check the implementation of the methods.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-state/src/main/java/io/getstream/chat/android/state/plugin/listener/internal/QueryChannelListenerState.kt [hunk 1]
```diff
@@ -49,6 +47,7 @@ internal class QueryChannelListenerState(private val logic: LogicRegistry) : Que
         channelId: String,
         request: QueryChannelRequest,
     ) {
+        kotlin.Result
```
Comment (skydoves) [inline stream-chat-android-state/src/main/java/io/getstream/chat/android/state/plugin/listener/internal/QueryChannelListenerState.kt:50]: Is this added by mistake? 😄

File: stream-chat-android-core/src/main/java/io/getstream/chat/android/client/utils/Result.kt [hunk 2]
```diff
@@ -35,6 +35,24 @@ public sealed class Result<out A : Any> {
...
         inline get() = this is Failure
 
+    /**
+     * Returns the encapsulated value if this instance represents [Success] [Result.isSuccess] or `null`
+     * if it is [Failure] [Result.isFailure].
+     */
+    public fun getOrNull(): A? = when (this) {
+        is Success -> value
+        is Failure -> null
+    }
```
Comment (skydoves) [inline stream-chat-android-core/src/main/java/io/getstream/chat/android/client/utils/Result.kt:45]: What do you think about adding `getOrThrow` as well, which is one of the popular extensions of the `Result` class?

**Review by skydoves — APPROVED** [review]
Looks good to me!
I added some minor comments.

File: stream-chat-android-core/src/main/java/io/getstream/chat/android/client/utils/Result.kt (hunk 2 above)
Comment (adasiewiczr) [reply stream-chat-android-core/src/main/java/io/getstream/chat/android/client/utils/Result.kt:45]: that is a good idea. Let me add it

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**